eslint.config.cjs

```diff
@@ -1,50 +1,194 @@
-const js = require("@eslint/js");
-const ts = require("@typescript-eslint/eslint-plugin");
-const tsParser = require("@typescript-eslint/parser");
-
+/* eslint-disable unicorn/no-useless-fallback-in-spread */
+/* eslint-disable no-undef */
 //-----------------------------------------------------------------------------
 // NOTE: The following plugins have been removed from the package.json file
 // eslint-plugin-oclif linter repo has been archived
 // eslint-config-oclif-typescript linter repo has been archived
 
-// Export the final config
+// -----------------------------------------------------------------------------
+// Eslint packages used:
+// "eslint" –> The core ESLint package. It is the base linter.
+// NOTE: Styles have moved from eslint to @stylistic/eslint-plugin (js and ts)
+//       Built-in stylistic rules for JavaScript and Typescript
+//       We are still using legacy config, so we need to install v3.x with
+//       npm i -D @stylistic/eslint-plugin@3
+// "@stylistic/eslint-plugin" -> Used for general stylistic rules, which can be
+//    applied to both JavaScript and TypeScript.
+// See here for additional information on ESLint Stylistic Stylistic Formatting
+// for ESLint https://eslint.style/
+//
+// "eslint-plugin-n" -> This package provides additional ESLint rules for Node.js
+//    Originally the these were provided by the eslint-plugin-node that it is
+//    no longer maintained
+//
+// "@eslint/js" –> This package provides core ESLint rules for JavaScript. It’s
+//    necessary to allow ESLint to enforce JavaScript best practices.
+//
+// "typescript-eslint" –> Tooling which enables the use of TypeScript with ESLint.
+//   This package is the entrypoint to consume additional tooling with ESLint.
+//   The package exports the following:
+//     config	 -> A utility function for creating type-safe flat configs
+//     configs ->	Shared ESLint (flat) configs
+//     parser	 -> A re-export of @typescript-eslint/parser
+//     plugin	 -> A re-export of @typescript-eslint/eslint-plugin
+// Notes: We still need to install the parser and plugin separately
```

Comment:
```suggestion
// TODO: We still need to install the parser and plugin separately
```

Should we make an issue for this in the repo?

Reply:
This is not a TODO, is done, I'm just noting here.